### ISSUE TITLE
strange error message about array types

### ISSUE THREAD

**Issue by mindplay-dk** [scrutinizer-ci/php-analyzer#788]
see https://scrutinizer-ci.com/g/mindplay-dk/walkway/inspections/2bd9ae58-e3f5-4062-a1d5-3088f3b9a747/issues/files/src/Module.php?status=existing&orderField=path&order=asc&issueId=390807d0-ff52-4a01-9731-fdf7692c8eea

```
it seems like array(&#039;int&#039; =>...&#039;[a-z0-9-]+&#039;) of type array<string,string,{"in...ring","slug":"string"}> is incompatible with the declared type array<integer,string> of property $symbols.
```

This error message looks like gibberish - I don't know what it means or why I'm getting a failed inspection here, it's type-hinted correctly as far as I can figure?

**Comment by jrfnl** [comment]
I'm no Scrutinizer expert, but just had a quick look at your code and I suspect the reason you're getting this error is because you **_documented_** the `$symbols` property as a string in line 51 of the same class.

In other words, the code isn't wrong, your documentation is. I suspect if you change that to the below, the error will disappear.
```php
@var array string[] map where symbol name => partial regular expression
```

**Comment by mindplay-dk** [comment]
> `@var array string[]`

You have two types in there? AFAIK, that's not a valid tag.

If I have this:

```php
/**
 * @var string[]
 */
public $symbols = [];
```

And I'm assigning this:

```php
$this->symbols = ['foo' => 'bar'];
```

Then then precise, inferred type being assigned is `array<string,string>` - which should be compatible with the declared type, `string[]` which is equivalent to the open type `array<?,string>`, which should be compatible with `array<string,string>` since you can't make any assertions about an open index type.

While I would have __loved__ to declare the type as the generic `array<string,string>`, this isn't supported by either Php Storm, nor by the de-facto php-doc standard, nor (sadly, though I [pushed hard](https://github.com/mindplay-dk/fig-standards/blob/features/generics/proposed/phpdoc.md#826-template) for this) by the upcoming php-doc PSR.

Does the (proprietary?) code analysis engine used by Scrutinizer support generics? Which code analysis tool is providing this inspection?

**Comment by gabrielsomoza** [comment]
I have a similar issue, [see this](https://scrutinizer-ci.com/g/baleen/migrations/inspections/fba4e868-7c58-4338-abf8-e1f6adec16fc/issues/files/src/Timeline/TimelineFactory.php?status=new&orderField=path&order=asc&issueId=d348818f-ce6b-4487-b2fa-2a8e5a85df49)

Basically, I have a function that accepts `VersionInterface[]|Collection` that passes those parameters through to another (internal) function that accepts the same parameter and same type-hints. I get the error message on the second function.

Maybe the analyzer doesn't keep track of types like `VersionInterface[]` and it just simplifies that to "array"? Any pointers?
